openai-assistant/create_assistant.py

```diff
@@ -3,57 +3,113 @@
+                        "enum": ["celsius", "fahrenheit"],
+                        "description": "The temperature unit to use. Infer this from the users location.",
+                    },
+                },
+                "required": ["location", "format"],
+            },
+        }
+    },
+    {
+        "type": "function",
+        "function": {
+            "name": "get_n_day_weather_forecast",
+            "description": "Get an N-day weather forecast",
+            "parameters": {
+                "type": "object",
+                "properties": {
+                    "location": {
+                        "type": "string",
+                        "description": "The city and state, e.g. San Francisco, CA",
+                    },
+                    "format": {
+                        "type": "string",
+                        "enum": ["celsius", "fahrenheit"],
+                        "description": "The temperature unit to use. Infer this from the users location.",
+                    },
+                    "num_days": {
+                        "type": "integer",
+                        "description": "The number of days to forecast",
+                    }
+                },
+                "required": ["location", "format", "num_days"]
+            },
+        }
+    },
+]
+
+
+def get_current_weather(input: dict):
+    # return dummy_weather
+    location = input.get("location")
+    format = input.get("format")
+    return "The current weather in {} is {} degrees {}".format(location, 20, format)
+
+
+def get_n_day_weather_forecast(input: dict):
+    # return dummy_weather
+    num_days = input.get("num_days")
+    location = input.get("location")
+    format = input.get("format")
+    return "The weather forecast for the next {} days in {} is {} degrees {}".format(num_days, location, 20, format)
+
+
+tool_map = {
+    "get_current_weather": get_current_weather,
+    "get_n_day_weather_forecast": get_n_day_weather_forecast,
+}
 
-print(tools)
 
 async def create():
-# ... your existing create function code ...
-   
+    # ... your existing create function code ...
+
     instructions = """You are a personal math tutor. Write and run code to answer math questions.
 Enclose math expressions in $$ (this is helpful to display latex). Example:
 
 Given a formula below $$ s = ut + \frac{1}{2}at^{2} $$ Calculate the value of $s$ when $u = 10\frac{m}{s}$ and $a = 2\frac{m}{s^{2}}$ at $t = 1s$
 """
-   
-    assistant = OpenAIAssistantRunnable.create_assistant(
-        name = "Math Tutor",
-        instructions = instructions,
-        tools = tools,
-        model = "gpt-4-1106-preview",
+
+    assistant = await client.beta.assistants.create(
+        name="Math Tutor",
+        instructions=instructions,
+        tools=tools,
+        model="gpt-4-1106-preview",
     )
-    print("langchain assistant created", assistant.assistant_id)
+    print("langchain assistant created", assistant.id)
```

Comment:
remove that print

Reply:
that print was there in the original cookbook but ok, ive removed it now